File: 2018/01/main.py
# Part 1
with open('in.txt', 'r') as f:
    ret = 0
    lines = f.readlines()
    for line in lines:
        ret += int(line)
    print(ret)


# Part 2
with open('in.txt', 'r') as f:
    ret = 0
    seen = set()
    lines = [int(l) for l in f.readlines()]
    while True:
        for line in lines:
            ret += line
            if ret in seen:
                print(ret)
                exit()
            seen.add(ret)

File: 2018/01/main.clj
; Part 1
(with-open [rdr (clojure.java.io/reader "in.txt")]
  (println (reduce + (map read-string (line-seq rdr)))))

; Part 2
(with-open [rdr (clojure.java.io/reader "in.txt")]
  (let [nums (map read-string (line-seq rdr))]
    (def curr 0)
    (def ret nil)
    (def seen #{})
    (while (nil? ret)
      (doseq [n nums]
        (def curr (+ n curr))
        (if (and (contains? seen curr) (nil? ret))
          ; how does one break out of a doseq?
          (def ret curr)
          (def seen (conj seen curr)))))
    (println ret)))
